### DISCUSSION TITLE
How do I access the currentUser email of AngularFireAuth?

### DISCUSSION THREAD

**Discussion by intheRain07** [angular/angularfire#2784] [Q&A]
I am a beginner of angular firenase.

After login, the currentUser is still null unless I use await and async, which disable accessing the currentUser in constructor. But I really need to init some variables in the constructors by the currentUser. I find that this is an issue for the new version of firebase, since many previous tutorials simply access the currentUser and its properities.

Here is my code:
```
constructor(
    public _angularFireAuth: AngularFireAuth
  ) { 
    }

this._angularFireAuth.currentUser.displayName;  // error: Property 'displayName' does not exist on type 'Promise<User>'.
```

**Comment by KingDarBoja** [comment]
You should do as follow:

```ts
@Component({
  selector: 'app-root',
  template: `
    <div *ngIf="user$ | async as user">
      <h1>Hello {{ user.displayName }}!</h1>
      <h2>Hello {{ user.email }}!</h2>
      <button (click)="logout()">Logout</button>
    </div>
  `,
})
class SomeComponent {
  user$ = this.auth.user;

  constructor(private readonly auth: AngularFireAuth) {}
}
```

Take a look at https://github.com/angular/angularfire/blob/master/docs/auth/getting-started.md

**Reply by rgant** [reply]
Also note the difference between `AngularFireAuth.user` and `AngularFireAuth.authState`:

`AngularFireAuth.user`: triggered when users were signed in, signed out, or when the user's ID token changed in situations such as token expiry or password change.

`AngularFireAuth.authState`: only triggered on sign-in or sign-out

https://firebase.google.com/docs/reference/js/firebase.auth.Auth#onauthstatechanged
